admin/src/js/services/message-queue.js

```diff
@@ -0,0 +1,337 @@
+      };
+
+      messages.forEach(function(message) {
+        if (message.sms) {
+          return;
+        }
+
+        var content = {
+          translationKey: message.scheduled_sms.translation_key,
+          message: message.scheduled_sms.content
+        };
+        var pseudoDoc = {
+          _id: message.record.id,
+          contact: message.contact,
+          fields: message.record.fields,
+          locale: message.record.locale
+        };
+
+        message.sms = MessageQueueUtils.messages.generate(
+          settings,
+          translate,
+          pseudoDoc,
+          content,
+          message.scheduled_sms.recipient,
+          message.context
+        )[0];
+      });
+
+      return messages;
+    };
+
+    var getTaskDisplayName = function(task) {
+      if (task.translation_key) {
+        return $translate.instant(task.translation_key, { group: task.group });
+      }
+
+      return task.type && task.type + (task.group ? ':' + task.group : '' ) || false;
+    };
+
+    var formatMessages = function(messages) {
+      return messages.map(function(message) {
+        return {
+          record: {
+            id: message.record.id,
+            reportedDate: message.record.reported_date
+          },
+          recipient: message.recipient && message.recipient.name || message.sms.to,
+          task: getTaskDisplayName(message.task),
+          state: message.task.state,
+          stateHistory: message.task.state_history,
+          content: message.sms.message,
+          due: message.due,
+          link: !!message.record.form
+        };
+      });
+    };
+
+    var getParams = function(tab, skip, limit, descending) {
+      var list = {
+        limit: limit || 25,
+        skip: skip || 0,
+        reduce: false
+      };
+
+      var count = {
+        reduce: true,
+        group_level: 1
+      };
+
+      var paging;
+      switch (tab) {
+        case 'scheduled':
+          paging = {
+            start_key: [tab, 0],
+            end_key: [tab, {}]
+          };
+          break;
+        case 'due':
+          paging = {
+            start_key: [tab, {}],
+            end_key: [tab, 0],
+            descending: true
+          };
+          break;
+        case 'muted':
+          if (descending) {
+            // get all past muted messages, descending from most recent
+            paging = {
+              start_key: [ tab, new Date().getTime() ],
+              end_key: [ tab, 0 ],
+              descending: true
+            };
+          } else {
+            paging = {
+              // get all future muted messages
+              start_key: [ tab, new Date().getTime() ],
+              end_key: [ tab, {} ]
+            };
+          }
+          break;
+      }
+
+      return {
+        list: Object.assign(list, paging),
+        count: Object.assign(count, paging)
+      };
+    };
+
+    return {
+      loadTranslations: function() {
+        return Languages()
```

Comment:
Instead of this can you use the [$translate.onReady](https://angular-translate.github.io/docs/#/api/pascalprecht.translate.$translate#methods_onready) function? It _should_ do the same thing.

Reply:
$translate.onReady is fired when the first translation is loaded, which would be already loaded by `MainCtrl` (https://github.com/medic/medic-webapp/blob/ce12674bb3961d91471bc1b65c35e54de58172ae/admin/src/js/controllers/main.js#L9) . It does not load all the languages, like it would have to.